static/app/utils/featureObserver.spec.ts

```diff
@@ -125,4 +133,119 @@ describe('FeatureObserver', () => {
+
+      project.features.includes('enable-proj-flag');
+      project.features.includes('replay-mobile-ui');
+      expect(inst.getFeatureFlags().values).toEqual([
+        {flag: 'feature.projects:enable-proj-flag', result: true},
+        {flag: 'feature.projects:replay-mobile-ui', result: false},
+      ]);
+
+      // do more evaluations to fill up and overflow the buffer
+      project.features.includes('enable-performance');
+      project.features.includes('autofix-ui');
+      project.features.includes('new-issue-details');
+      expect(inst.getFeatureFlags().values).toEqual([
+        {flag: 'feature.projects:enable-performance', result: true},
+        {flag: 'feature.projects:autofix-ui', result: false},
+        {flag: 'feature.projects:new-issue-details', result: false},
+      ]);
+    });
+
+    it('should not change the functionality of `includes`', () => {
+      const inst = new FeatureObserver({bufferSize: 3});
+      inst.observeProjectFlags({project});
+      expect(inst.getFeatureFlags().values).toEqual([]);
+
+      project.features.includes('enable-proj-flag');
+      project.features.includes('replay-mobile-ui');
+      expect(inst.getFeatureFlags().values).toEqual([
+        {flag: 'feature.projects:enable-proj-flag', result: true},
+        {flag: 'feature.projects:replay-mobile-ui', result: false},
+      ]);
+
+      expect(project.features.includes('enable-proj-flag')).toBe(true);
+      expect(project.features.includes('replay-mobile-ui')).toBe(false);
+    });
+  });
+
+  describe('observeProjectFlags and observeOrganizationFlags', () => {
+    it('should add recently evaluated org and proj flags to the flag queue', () => {
+      const inst = new FeatureObserver({bufferSize: 3});
+      expect(project.features).toEqual(['enable-proj-flag', 'enable-performance']);
+      expect(organization.features).toEqual([
+        'enable-issues',
+        'enable-profiling',
+        'enable-replay',
+      ]);
+
+      inst.observeProjectFlags({project});
+      inst.observeOrganizationFlags({organization});
+      expect(inst.getFeatureFlags().values).toEqual([]);
+
+      project.features.includes('enable-proj-flag');
+      project.features.includes('enable-replay');
+      organization.features.includes('enable-issues');
+      expect(inst.getFeatureFlags().values).toEqual([
+        {flag: 'feature.projects:enable-proj-flag', result: true},
+        {flag: 'feature.projects:enable-replay', result: false},
+        {flag: 'feature.organizations:enable-issues', result: true},
+      ]);
+
+      organization.features.includes('enable-replay');
+      expect(inst.getFeatureFlags().values).toEqual([
+        {flag: 'feature.projects:enable-replay', result: false},
+        {flag: 'feature.organizations:enable-issues', result: true},
+        {flag: 'feature.organizations:enable-replay', result: true},
+      ]);
+    });
+
+    it('should keep the same queue if the project changes', () => {
```

Comment:
@ryan953 test to check that the queue stays the same if we switch to a different project